File: src/main/java/org/minturtle/careersupport/interview/repository/InterviewTemplateRepository.java
package org.minturtle.careersupport.interview.repository;

import org.minturtle.careersupport.common.aop.Logging;
import org.minturtle.careersupport.interview.entity.InterviewTemplate;
import org.springframework.data.domain.Pageable;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
import reactor.core.publisher.Flux;


@Logging
public interface InterviewTemplateRepository extends ReactiveMongoRepository<InterviewTemplate, String> {
    Flux<InterviewTemplate> findByUserId(String userId, Pageable pageable);
}

File: src/main/kotlin/org/minturtle/careersupport/interview/repository/InterviewTemplateRepository.kt
package org.minturtle.careersupport.interview.repository

import kotlinx.coroutines.flow.Flow
import org.minturtle.careersupport.interview.entity.InterviewTemplate
import org.springframework.data.domain.Pageable
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import org.springframework.data.repository.kotlin.CoroutineCrudRepository
import reactor.core.publisher.Flux

interface InterviewTemplateRepository: CoroutineCrudRepository<InterviewTemplate, String> {
    suspend fun findByUserId(userId: String, pageable: Pageable): Flow<InterviewTemplate>
}